src/main/java/org/moonlightcontroller/managers/XidGenerator.java

```diff
@@ -0,0 +1,22 @@
+package org.moonlightcontroller.managers;
+
+public class XidGenerator {
+	private static int xid = 10000;
+	private static XidGenerator instance;
+	
+	private XidGenerator() {
+		
+	}
+	
+	public static XidGenerator getInstance() {
+		if (instance == null) {

```

Comment:
This better be thread safe, e.g., wrap the if with a "synchronized(XidGenerator.class)" block and then wrap the synchronized block with another if (and keep the return outside everything).